mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -353,6 +353,43 @@ class ConstructorPattern final
   }
 };
 
+//===----------------------------------------------------------------------===//
+// MethodPattern - Converts SYCLMethodOpInterface to LLVM.
+//===----------------------------------------------------------------------===//
+
+static LogicalResult convertMethod(SYCLMethodOpInterface op,
+                                   ConversionPatternRewriter &rewriter) {
+  LLVM_DEBUG(llvm::dbgs() << "MethodPattern: Rewriting op: "; op.dump();
+             llvm::dbgs() << "\n");
+
+  SmallVector<mlir::Value> methodArgs(op->getOperands());
+
+  if (op.getBaseType() != methodArgs[0].getType()) {
+    auto Cast = rewriter.create<SYCLCastOp>(op.getLoc(), op.getBaseType(),
+                                            op->getOperand(0));
+    methodArgs[0] = Cast;
+  }
```

Comment:
```suggestion
  if (op.getBaseType() != methodArgs[0].getType())
    methodArgs[0] = rewriter.create<SYCLCastOp>(op.getLoc(), op.getBaseType(),
                                            op->getOperand(0));
```

Reply:
Why does this transformation need to add a Cast ? I would expect the first operand of the method to already have the correct type.

Reply:
In some cases, the function will be a member of a base class. As we are abstracting that cast to the base class when inserting the new operations, we have to reinsert them here.